core/tx_pool.go

```diff
@@ -823,7 +823,20 @@ func (pool *TxPool) validateStakingTx(tx *staking.StakingTransaction) error {
 			return errors.WithMessagef(ErrInvalidSender, "staking transaction sender is %s", b32)
 		}
 
-		_, _, err = VerifyAndDelegateFromMsg(pool.currentState, stkMsg)
+		chain, ok := pool.chain.(ChainContext)
+		if !ok {
+			return nil // for testing, chain could be testing blockchain
```

Comment:
what does this mean here? any info/log needed?

Reply:
This part is copied from CollectReward. I think it's for unittests. Added log.